neo/SmartContract/NativeContract/NativeNep5Neo.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Neo.Ledger;
+using Neo.VM;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Neo.SmartContract.NativeContract
+{
+    class NativeContract_Nep5Neo : INativeContract
+    {
+        public string Name => "Nep5Neo";
+
+        public long Price => 1024;
+
+        public ContractParameterType[] Parameter_list => new ContractParameterType[] {
+            ContractParameterType.String,
+            ContractParameterType.Array
+        };
+
+        public ContractParameterType Return_type => ContractParameterType.ByteArray;
+
+        public ContractPropertyState Contract_properties => ContractPropertyState.HasStorage | ContractPropertyState.Payable;
+
+        public string Version => "";
+
+        public string Author => "";
+
+        public string Email => "";
+
+        public string Description => "";
+
+
+        public bool Contract_Main(ExecutionEngine engine)
+        {
+            //throw new NotImplementedException();
+            var trigger = (engine.Service as StandardService).Trigger;
+            if (trigger == TriggerType.Verification)
+            {
+                StackItem returnvalue = false;
+                engine.CurrentContext.EvaluationStack.Push(returnvalue);
+                return true;
+            }
+            else if (trigger == TriggerType.Application)
```

Comment:
It would be nice if we had some standard for contract tokens that could allow sending coins using attribute parameters for `from`, `to`, and `amount`, and that mechanism require bringing the contract on as a witness so we could use the verification trigger to cause the transfer to fail if not enough balance is available up front. For simple sends by wallets, it would be nice if wallets used this mechanism for sending so the blockchain does not record useless transactions from attempts to spend funds that are not available.

Reply:
Amazing suggestion Jeff! It turns out to be quite easy to do and quite compatible to existing NEP-5 tokens.
On `transfer` method we just add the following line:
```cs
CheckWitness( Hash160( 
       CreateScript($"PUSH({value.ToArray()});PUSH('balance');GT;THROWIFNOT;" )
                       ) );
```
So, native contracts can start requiring wallets to pre-verify enough funds on transaction, and we can easily extend this to all NEP-5 tokens, since it won't break any of them.

Reply:
Jeff, I concluded this does not affect standard. People can use this a good practice, during verification.